tests/Doctrine/Tests/ORM/Functional/SecondLevelCacheManyToManyTest.php

```diff
@@ -188,12 +188,10 @@ public function testStoreManyToManyAssociationWhitCascade() : void
         self::assertEquals($queryCount1, $this->getCurrentQueryCount());
     }
 
-    /**
-     * @expectedException \Doctrine\ORM\Cache\Exception\CacheException
-     * @expectedExceptionMessage Cannot update a readonly collection "Doctrine\Tests\Models\Cache\Travel#visitedCities
-     */
     public function testReadOnlyCollection() : void
     {
+        $this->expectException('Doctrine\ORM\Cache\Exception\CacheException');
```

Comment:
Which method is throwing this exception? Please move the `expectException*()` methods before the affected code.